### ISSUE TITLE
Why isn't my .bib updating?

### ISSUE THREAD

**Issue by paula-j-m** [retorquere/zotero-better-bibtex#1660]
*Please verify **all** of the below before submitting*:

1. you have picked the right category for the issue in the previous screen.
2. You are on the latest release of Zotero
3. in the Zotero addons screen you can see that you have the latest release of BBT (https://github.com/retorquere/zotero-better-bibtex/releases/latest)
4 you are posting a single question.
5. the issue has a subject that succinctly describes the problem or question.
6. you are available for follow-up questions and testing.

Picking the right issue category is really important. Each category (`Export`, `General error`, `Import`, `Key generation`, `Question`) has different instructions for gathering the data necessary required to resolve the issue you are experiencing

**Your question/suggestion:**

Hello- I am very new to using this program with Latex and Zotero(working on a dissertation). As I am working on my draft, I sometimes realize I haven't added all the citations I need to my collection on Zotero. So I add the citation, find the citation key, and add it into my latex file. then, usually about 10 minutes later, I compile Latex-bibtex-latex-latex, and so far only 1/5 or so times has the bibliography updated, despite the fact that in preferences I have it maked "on change". Yesterday I was stuck for several hours trying to figure this out- I never got it to update automatically, and couldn't get a .bib file to export from that collection that included all of my items. Today the first time I added a new citation it worked fine, and the second two have not updated. If I export the collection to "create bibliography" or to a new location, the list is complete, but if I go into preferences and select "update now" and then check the .bib file, they are still missing. 

I hope I am just missing something basic here but I have not been able to figure it out!

**Comment by retorquere** [comment]
Please go to the [new issue](https://github.com/retorquere/zotero-better-bibtex/issues/new/choose) page, click the "get started" button next to 

> **Export bug/enhancement/question**
> You appear to have found a bug in BBTs export, are proposing an enhancement to the exports, or have a question about exporting items.

and follow the instructions on the page that then appears. I need a debug log as described there, I cannot begin to guess at the problem without that.

**Comment by paula-j-m** [comment]
Thank you for the quick response- as expected it was a user error that had
to do with spaces in the name of the file and if it is something more
substantial then I will follow your instructions below.

**Comment by retorquere** [comment]
Spaces in the name should not be a problem. I'd still consider that a bug. But I'd need a debug log for when you updated an item and it failed to update the .bib file.

**Comment by paula-j-m** [comment]
Right, sorry I wasn't clear, I realize now that there was not an issue with
the export, but with me not realizing that I had two different files (one
with a space, one without), and that when I changed the file name after the
export, then (of course), it didn't update anymore from zotero. When I
thought it was working, it was when I did a fresh export, then just checked
the .bib file after I added some new references, and it was fine. In my
attempt to use it in my actual document, then I had the wrong file name and
changing the file name only updated me to the most recent export. - I
figured I'd write it out in case someone else out there needs the very
basic reminder that if you change the file name after you export, it will
not be the same file and will not update!
